Source/CarthageKit/Project.swift

```diff
@@ -684,50 +698,85 @@ public final class Project {
 			.then(SignalProducer<(), CarthageError>.empty)
 	}
 
+	/// - parameters:
+	/// 	- flatMapTransform: defaults to identity function. Only ever used for one thing, but parameterized for unit testing purposes.
+	private func fileSystemObjects(for dependency: Dependency<PinnedVersion>, withRepository repositoryURL: URL, at pathList: String, flatMapTransform: @escaping (String) -> String? = { $0 }) -> SignalProducer<[String], CarthageError> {
```

Comment:
This seems like it should be return a `SignalProducer<String, CarthageError>` and sending multiple values. It should only send `[String]`s if it expects to send multiple arrays. I don't think that's the case here.